src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -619,15 +619,17 @@
   ;; will then cause the pool connection to throw a (generic)
   ;; SQLException.
   (log/info (trs "Ensuring {0} database is up to date" db-name))
-  (let  [migrator (:migrator-username db-config)]
+  (let  [connection-migrator (:connection-migrator-username db-config)
+         migrator-password (:migrator-password db-config)
+         migrator (:migrator-username db-config)]
     (with-open [db-pool (-> (assoc db-config
                                    :pool-name (if db-name
                                                 (str "PDBMigrationsPool: " db-name)
                                                 "PDBMigrationsPool")
                                    :connection-timeout 3000
                                    :rewrite-batched-inserts "true"
-                                   :user migrator
-                                   :password (:migrator-password db-config))
+                                   :user connection-migrator
```

Comment:
@rbrw 
Maybe we can achieve the same thing by changing https://github.com/puppetlabs/puppetdb/blob/2d547241c619b6531dd5e98597add84d27ef6f17/src/puppetlabs/puppetdb/jdbc.clj#L642